src/query/graphite/native/aggregation_functions.go

```diff
@@ -94,6 +94,36 @@ func maxSeries(ctx *common.Context, series multiplePathSpecs) (ts.SeriesList, er
 	return combineSeries(ctx, series, wrapPathExpr(maxSeriesFnName, ts.SeriesList(series)), ts.Max)
 }
 
+// medianSeries takes a list of series and returns a new series containing the
+// median value across the series at each datapoint
+func medianSeries(ctx *common.Context, series multiplePathSpecs) (ts.SeriesList, error) {
+	if len(series.Values) == 0 {
+		return ts.NewSeriesList(), nil
+	}
+	normalized, start, end, millisPerStep, err := common.Normalize(ctx, ts.SeriesList(series))
+	if err != nil {
+		return ts.NewSeriesList(), err
+	}
+	numSteps := ts.NumSteps(start, end, millisPerStep)
+	values   := ts.NewValues(ctx, millisPerStep, numSteps)
+
+
+	for i := 0; i < numSteps; i++ {
+		valuesAtTime := []float64{}
+		for _, series := range normalized.Values {
+			valuesAtTime = append(valuesAtTime, series.ValueAt(i))
+		}
+		values.SetValueAt(i, ts.Median(valuesAtTime, len(valuesAtTime)))
+	}
```

Comment:
Best to declare this on line 110 with the size we know we'll use each time. That way can use the same slice each time in the inner loop and there can be a single allocation instead of an array allocation per datapoint step processed.

e.g.
```suggestion
	valuesAtTime := make([]float64, len(normalized.Values))
	for i := 0; i < numSteps; i++ {
		for j, series := range normalized.Values {
			valuesAtTime[j] = series.ValueAt(i)
		}
		values.SetValueAt(i, ts.Median(valuesAtTime, len(valuesAtTime)))
	}
```